### PR TITLE
Fix genesis block creation logic

### LINKED ISSUE #147: Fix genesis block creation logic
### Expected behavior
Genesis block should be created based on `lisk-core` instance

### Actual behavior
Currently, genesis block is created based on default application instance. This results in the error when starting node with migrated genesis block:
```
2023-08-29T08:05:34.629Z INFO Himanshus-MBP.fritz.box application 85680 Registered legacy module has command registerKeys
2023-08-29T08:05:34.646Z INFO Himanshus-MBP.fritz.box application 85680 [stateRoot=c87b4dca0c82e31455f28984322c189b33c19216ccae19be5f6a5c90a7201682 height=20171933] Start caching genesis state
    Error: Invalid state root `Not matching with expected`
2023-08-29T08:05:37.445Z INFO Himanshus-MBP.fritz.box application 85680 [errorCode=2 message=process.exit] Application shutdown started
```
### Steps to reproduce

- Create genesis block using migrator (`testnet` or `mainnet`)
- Start node with the migrated genesis block
- Ref https://github.com/LiskHQ/lisk-sdk/issues/8909

### Which version(s) does this affect? (Environment, OS, etc...)
v2.0.0-beta.2

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?

This PR resolves #147 

### How was it solved?
- [x] Fix genesis block creation logic to create genesis block with `lisk-core` instance
- [x] Node should start with the genesis block created
- [x] Fix node auto start logic

### How was it tested?
Locally against `testnet`

### CODE REVIEW

File: src/utils/genesis_block.ts [hunk 1]
```diff
@@ -46,45 +44,26 @@ export const createChecksum = async (filePath: string): Promise<string> => {
...
-
-	const genesisBlock = await app.generateGenesisBlock(input);
-	genesisBlock.header.version = SNAPSHOT_BLOCK_VERSION;
-	return genesisBlock;
+) => {
+	const timestamp = blockAtSnapshotHeight.header.timestamp + snapshotTimeGap;
+	const height = blockAtSnapshotHeight.header.height + 1;
+	const previousBlockID = blockAtSnapshotHeight.header.previousBlockID.toString('hex');
+	const genesisBlockCreateCommand = `lisk-core genesis-block:create --network ${network} --config=${config} --output=${output} --assets-file=${output}/genesis_assets.json --height=${height} --previous-block-id=${previousBlockID} --timestamp=${timestamp}`;
+	await execAsync(genesisBlockCreateCommand);
```
Comment (sameersubudhi) [inline src/utils/genesis_block.ts:57]:
```suggestion
	const height = blockAtSnapshotHeight.header.height + 1;
	const timestamp = blockAtSnapshotHeight.header.timestamp + snapshotTimeGap;
	const previousBlockID = blockAtSnapshotHeight.header.id.toString('hex');

	const genesisBlockCreateCommand = `lisk-core genesis-block:create --network ${network} --config=${configFilepath} --output=${outputPath} --assets-file=${outputPath}/genesis_assets.json --height=${height} --previous-block-id=${previousBlockID} --timestamp=${timestamp}`;

	await execAsync(genesisBlockCreateCommand);
```

File: src/utils/genesis_block.ts [hunk 2]
```diff
@@ -46,45 +44,26 @@ export const createChecksum = async (filePath: string): Promise<string> => {
 };
 
 export const createGenesisBlock = async (
-	// TODO: Update type any once GenesisBlockGenerateInput exported by SDK
-	app: any,
-	assets: GenesisAssetEntry[],
+	network: string,
+	config: string,
+	output: string,
```
Comment (sameersubudhi) [inline src/utils/genesis_block.ts:49]:
```suggestion
	network: string,
	configFilepath: string,
	outputPath: string,
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please address the following and the remaining changes discussed offline.

**Review by sameersubudhi — APPROVED** [review]

**Review by shuse2 — APPROVED** [review]
other than one minor comment it looks good

File: src/utils/node.ts [hunk 3]
```diff
@@ -22,14 +24,18 @@ import { Port } from '../types';
 import { getAPIClient } from '../client';
 import { DEFAULT_PORT_P2P, DEFAULT_PORT_RPC } from '../constants';
 
-// TODO: Remove custom registry after Lisk Core is published to public NPM registry
-const INSTALL_LISK_CORE_COMMAND = 'npm i -g lisk-core --registry https://npm.lisk.com';
+const INSTALL_LISK_CORE_COMMAND = 'npm i -g lisk-core@^4.0.0-beta.5';
```
[the remark this answers is not in the document]
Comment (sameersubudhi) [reply src/utils/node.ts:27]: We'll update it to rc, when we update the SDK dependency with v6.0.0-rc.0 👍

**Pull request merged**